File: 351003/Zarenok/demo/publisher/src/main/java/com/example/demo/repository/MarkRepository.java
package com.example.demo.repository;

import com.example.demo.model.Mark;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface MarkRepository extends JpaRepository<Mark, Long>, JpaSpecificationExecutor<Mark> {
    boolean existsByName(String name);
    Optional<Mark> findByName(String name);
}

File: 351003/Soika/entities-app/publisher/src/main/kotlin/com/example/entitiesapp/repository/MarkRepository.kt
package com.example.entitiesapp.repository

import com.example.entitiesapp.model.Mark
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface MarkRepository : JpaRepository<Mark, Long> {
    fun findByName(name: String): Mark?
}